integer constant overflow on gccgo and 32-bit

After upgrading to the latest version on v2, I'm running into this error when building with gccgo.
```
[2016/08/02 19:13:01.577] # gopkg.in/mgo.v2/bson
[2016/08/02 19:13:01.577] vendor/src/gopkg.in/mgo.v2/bson/json.go:320:11: error: integer constant overflow
[2016/08/02 19:13:01.577]   if n <= 1<<53 {
```

Perhaps `int64` should be used explicitly here instead of `int`.


*Update:* This issue is also occurring using the standard go compiler on 32-bit Windows.
```
[2016/08/02 21:21:22.468] # gopkg.in/mgo.v2/bson
[2016/08/02 21:21:22.468] .gopath\src\gopkg.in\mgo.v2\bson\json.go:320: constant 9007199254740992 overflows int
```

--- a/bson/json.go
+++ b/bson/json.go
@@ -317,7 +317,7 @@ func jencNumberLong(v interface{}) ([]byte, error) {
 func jencInt(v interface{}) ([]byte, error) {
 	n := v.(int)
 	f := `{"$numberLong":"%d"}`
-	if n <= 1<<53 {
+	if int64(n) <= 1<<53 {
 		f = `%d`
 	}
 	return fbytes(f, n), nil

